Sources/SwiftBSON/BSONEncoder.swift

```diff
@@ -760,79 +760,61 @@ extension _BSONEncoder: SingleValueEncodingContainer {
     }
 }
 
+internal protocol BSONRepresentable {
+    func toBSON() throws -> BSON
```

Comment:
this is marked as throws (even though all current implementations do not require throwing), because it will need to start throwing as part of the null bytes fix. (see below)